test/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests.csproj

```diff
@@ -1,10 +1,10 @@
 <Project Sdk="Microsoft.NET.Sdk">
+
   <PropertyGroup>
-    <Description>Unit test project for OpenTelemetry StackExchangeRedis instrumentation</Description>
     <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
-    <TargetFrameworks Condition="$(TARGET_FRAMEWORK) == ''">$(SupportedNetTargets)</TargetFrameworks>
-    <TargetFrameworks Condition="$(TARGET_FRAMEWORK) == '' and $(OS) == 'Windows_NT'">$(TargetFrameworks);net462</TargetFrameworks>
-    <TargetFrameworks Condition="$(TARGET_FRAMEWORK) != ''">$(TARGET_FRAMEWORK)</TargetFrameworks>
```

Comment:
Please check test\OpenTelemetry.Instrumentation.StackExchangeRedis.Tests\Dockerfile

Reply:
In fact it's the same for `Instrumentation.ConfluentKafka` and the csproj is much simpler. The CI is green. Do you still want me to revert?

Reply:
It will be great to understand how it is used, and if it will be needed together with .NET9.
I do not remember this details to be honest.

Reply:
After investigation, I think the goal is to let the dockerfile override the TFM by setting TARGET_FRAMEWORK env. If it is not set, there is a fallback to $(SupportedNetTargets). We need to understand if it is really working as expected for ConfluentKafka because this logic is not in the csproj.

Reply:
Short version: It is safe to remove 

Long version:

I had to go way way back to dig this up 😄

https://github.com/open-telemetry/opentelemetry-dotnet/pull/1266#discussion_r490638648

But that was for a docker file which had:

```
FROM mcr.microsoft.com/dotnet/core/sdk:${SDK_VERSION} AS build
FROM mcr.microsoft.com/dotnet/core/sdk:${SDK_VERSION} AS final
```

At some point it looks like we switched to:

```
FROM mcr.microsoft.com/dotnet/sdk:${BUILD_SDK_VERSION} AS build
FROM mcr.microsoft.com/dotnet/sdk:${TEST_SDK_VERSION} AS final
```

So we now have a different SDK version for build than for test execution. I think that is what made it unnecessary.

We have these compose files FYI where those vars are set:

* https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/build/docker-compose.net6.0.yml

* https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/build/docker-compose.net8.0.yml

We should be able to remove that `net6.0.yml` one once we have removed `net6.0` everywhere.